Assignments/Program.cs

```diff
@@ -1 +1,116 @@
+      Console.WriteLine ("\x1B[4m" + "Numbers to Words:" + "\x1B[0m");
+      for (int i = 0; i < wTestCases.Count; i++) {
+         var item = wTestCases[i];
+         Console.Write ($"{i + 1,2}) {item,9} ==> ");
+         WriteInYellow (ToWords (item));
+         Console.ResetColor ();
+      }
+      Console.WriteLine ("\n\x1B[4m" + "Numbers to Romans:" + "\x1B[0m");
+      for (int i = 0; i < rTestCases.Count; i++) {
+         var item = rTestCases[i];
+         Console.Write ($"{i + 1,2}) {item,4} ==> ");
+         WriteInYellow (ToRomans (item));
+         Console.ResetColor ();
+      }
+   }
+
+   /// <summary>Converts input number to equivalent words</summary>
+   public static string ToWords (int num) {
+      if (num == 0) return "Zero";
+      string input = num.ToString ().PadLeft (9, '0'),
+            output = "";
+      List<string> ones = new () { "", "One", "Two", "Three", "Four", "Five", "Six", "Seven", "Eight", "Nine", "Ten",
+      "Eleven", "Twelve", "Thirteen", "Fourteen", "Fifteen", "Sixteen", "Seventeen", "Eighteen", "Nineteen" },
+                   tens = new () { "", "Ten", "Twenty", "Thirty", "Forty", "Fifty", "Sixty", "Seventy", "Eighty", "Ninety" };
+      for (int i = 0; i < 3; i++) {
+         int part = int.Parse (input.Substring (i * 3, 3));
+         State s = State.A;
+         for (int loop = 3; loop > 0; loop--) {
+            int n = part / (int)Math.Pow (10, loop - 1) % 10, d3 = part % 10;
+            switch (s, n) {
+               case (State.A, >= 0 and <= 9): {
+                     s = State.B;
+                     if (n != 0) {
+                        output += ones[n] + " Hundred ";
+                        if (part / 10 % 10 > 0 || d3 > 0) output += "and ";
+                     }
+                     break;
+                  }
+               case (State.B, 1): {
+                     s = State.D;
+                     output += ones[10 + d3] + " ";
+                     break;
+                  }
+               case (State.B, 0 or (>= 2 and <= 9)): {
+                     s = State.C;
+                     if (n != 0) output += tens[n] + ((i == 2 && d3 > 0) ? "-" : " ");
+                     break;
+                  }
+               case (State.C, >= 0 and <= 9): {
+                     s = State.D;
+                     if (n != 0) output += ones[n] + " ";
+                     break;
+                  }
+               default: { s = State.D; break; }
+            }
+         }
+         if (part > 0 && i != 2) {
+            var (cond, key) = (i == 0) ? (num / 1000 % 1000 > 0 || num % 1000 > 0, "Million") : (num % 1000 > 0, "Thousand");
+            output += key + (cond ? ", " : " ");
+         }
+      }
+      return output;
+   }
+   enum State { A, B, C, D };
+   #region State Machine Diagram for converting any 3-digit Number to Words:

```

Comment:
Instead you could have added a png to the project and embed it as a resource. This doesn't give a clear picture though.